tests/driver_dsp0401/main.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the DSP0401 4 digit display.
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <inttypes.h>
+
+#include "dsp0401_params.h"
+#include "dsp0401.h"
+#include "xtimer.h"
+
+#define TEST_DELAY      (2U)   /* 2 seconds delay between each test */
+#define WORD_DELAY      (750U) /* 750 milliseconds delay between each word */
+#define SCROLL_DELAY    (200U) /* 200 milliseconds delay between character shift */
+#ifndef LOOPS
+#define LOOPS           (3U)   /* Number of display loops before exit */
+#endif
+
+int main(void)
+{
+    dsp0401_t dev;
+
+    puts("Starting DSP0401 test application\n");
+
+    puts("Initializing DSP4010 device.");
+    int result = dsp0401_init(&dev, &dsp0401_params[0]);
+    switch (result) {
```

Comment:
this looks rather redundant, if there is an error -> enable debug in the driver and it will print the cause.

This switch case unnecessarily bloats the test binary size, which may exclude small boards.

I'd recommend to reduce this to:

```
if (dsp0401_init(&dev, &dsp0401_params[0]) != 0) {
    puts("[Error] Failed to init display!");
    return 1;
}
```